### ISSUE TITLE
CA2000 false positives with out parameters

### ISSUE THREAD

**Issue by jherby2k** [dotnet/roslyn-analyzers#2245]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.0

#### Diagnostic ID

[CA2000](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope)

#### Repro steps

1. Try to dispose a bunch of disposable objects from a concurrent queue:

```c#
while (Handles.TryDequeue(out var handle))
                handle.Dispose();
```

-or-

1. Try to dispose a handle obtained via an out parameter using the IDisposable pattern

```c#
readonly CustomNativeHandle _handle;

internal TestConstructor()
{
    NativeMethods.GetHandle(out _handle);
}

public void Dispose()
{
    _handle.Dispose();
}
```

#### Expected behavior

No warning - both of these dispose of handles correctly, AFAICT. No warnings under last stable release.

#### Actual behavior

CA2000 warning

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/2348
